React/Recipes/How to build core component.md

```diff
@@ -0,0 +1,639 @@
+
+To solve this component we need to shift our mindset from vertical (page based) "slicing" flow to horizontal flow.
+
+<figure>
+  <img src="/handbook/img/frontend/react-recipes/vertical_vs_horizontal.svg">
+  <figcaption style="text-align: center">
+    <a href="/handbook/img/frontend/react-recipes/vertical_vs_horizontal.svg" >Vertical vs Horizontal Flow</a>
+  </figcaption>
+</figure>
+
+To do this we need to focus more on Design System and reusability of our components.
+If your design doesn't have any Design System you should ask designer to make on. Here is the example of [Chakra UI Design System](https://www.figma.com/community/file/971408767069651759) made in Figma.
+
+### Card design
+
+Here is the simple example of two different card designs, `user` card in the first row and the `job` card.
+
+![Cards](/img/react-recipes/cards.png)
+
+### Multiple similar cards
+
+First step is to bootstrap the layout with inline props.
+
+```tsx
+// ./src/components/shared/user/UserCard/UserCard.tsx
+
+import * as React from "react";
+import { Box, Heading, Image, Text } from "@chakra-ui/react";
+import { FC } from "react";
+import { IUser } from "../../../../interfaces/IUser";
+
+export interface IUserCardProps {
+  user: IUser;
+}
+
+export const UserCard: FC<IUserCardProps> = ({ user }) => (
+  <Box borderRadius="md" border="1px" borderColor="gray.200">
+    <Image
+      src={user.picture}
+      borderRadius="full"
+      w="150px"
+      h="150px"
+      fit="cover"
+      mb="4"
+      mt="4"
+      mx="auto"
+      bg="gray.200"
+    />
+    <Heading as="h1" size="sm" px="5" mb="4" textAlign="center">
+      {`${user.name} ${user.surname}`}
+    </Heading>
+    <Text fontSize="xs" px="5" mb="5" noOfLines={3} textAlign="center">
+      {user.bio}
+    </Text>
+  </Box>
+);
+```
+
+```tsx
+// ./src/components/shared/user/JobCard/JobCard.tsx
+
+import * as React from "react";
+import { Box, Heading, Image, Text } from "@chakra-ui/react";
+import { FC } from "react";
+import { IJob } from "../../../../interfaces/IJob";
+
+export interface IJobCardProps {
+  job: IJob;
+}
+
+export const JobCard: FC<IJobCardProps> = ({ job }) => (
+  <Box borderRadius="md" border="1px" borderColor="gray.200">
+    <Image
+      src={job.thumbnail}
+      borderTopRadius="md"
+      w="100%"
+      h="200px"
+      fit="cover"
+      mb="4"
+    />
+    <Heading as="h1" size="sm" px="5" mb="4">
+      {job.title}
+    </Heading>
+    <Text fontSize="xs" px="5" pb="5" noOfLines={3}>
+      {job.description}
+    </Text>
+  </Box>
+);
+```
+
+### When this is OK?
+
+When we are building a small one-off components which doesn't contain a lot of code and  are not shared across components.  
+
+### Why this approach can become a problem?
+
+This example is not so bad as it seems at firs because we don't have to name components (naming is hard), but as this component grows and new features are added, inline style will become a problem.
```

Comment:
```suggestion
This example is not as bad as it seems at first because we don't have to name components (naming is hard), but as this component grows and new features are added, the inline style will become a problem.
```